Small fixes to make it work with Python 3

kartikmohta opened the pull request:
Note that two tests fail due to internal changes in Python 3:
- test_iterable_literals_eval: The order of keys in dict changed in Py3
- test_no_evaluation: ast.literal_eval('5 -2') gives 3 in Py3

kartikmohta committed:
Small fixes to make it work with Python 3

Note that two tests fail due to internal changes in Python 3:
- test_iterable_literals_eval: The order of keys in dict changed in Py3
- test_no_evaluation: ast.literal_eval('5 -2') gives 3 in Py3

rhaschke commented:
~~I'm wondering whether these changes are really neccessary. Did you have specific issues running xacro with python3? Running the unittests on my side, I don't see any issues.~~
Sorry, the unittests didn't yet used python3 on my side. Now, I'm getting some errors too ;-)

kartikmohta commented:
`cStringIO` and `dict.iteritems` are not present in Python 3, so these changes are required.

rhaschke committed:
fixed minidom import on python3

rhaschke committed:
unittest: handle different dict ordering in python3

rhaschke commented on src/xacro/__init__.py:
```diff
@@ -950,7 +950,7 @@ def process_doc(doc,
         warning("Document is incompatible to --inorder processing.")
         warning("The following properties were redefined after usage:")
-        for k, v in symbols.redefined.iteritems():
+        for k, v in symbols.redefined.items():
```
Although iteritems() is more efficient than items() in python2, I think for the sake of simplicity, we can switch to items().

rhaschke commented on xacro.py:
```diff
@@ -50,7 +50,7 @@
 os.chdir(cur_dir)
 # Remove this dir from path
-sys.path = filter(lambda a: a not in [this_dir, this_dir_cwd], sys.path)
+sys.path = [a for a in sys.path if a not in [this_dir, this_dir_cwd]]
```
This is definitely more pythonic ;-)

rhaschke reviewed (changes requested):
Additionally to your requested changes, I had to
import xml.dom.minidom
in xmlutils.py. Strange, that you didn't notice that issue.

kartikmohta committed:
Add xml.dom.minidom import

rhaschke committed:
fixed literal parsing

Parsing of number and boolean literals was done with ast.literal_eval().
However, since python3 this function evaluates expressions like "5 -3"
as the number result and not a string anymore.
Hence resort to explicit type-casting to int, float, or boolean.

rhaschke commented on src/xacro/xmlutils.py:
```diff
@@ -31,6 +31,7 @@
 # Maintainer: Morgan Quigley <morgan@osrfoundation.org>
 
 import xml
```
The "import xml" can be removed then.

rhaschke commented:
As soon as #161 is merged, I will add some more commits, cleanup and re-issue as a new PR.
I fixed everything for python3 already, such that the unittests pass:
https://travis-ci.org/ubi-agni/xacro
